### ISSUE TITLE
Rule violations not detected for procedural code

### ISSUE THREAD

**Issue by james-a-henderson** [phpmd/phpmd#681]
- PHPMD version: 2.7.0
- PHP Version: 7.0.30
- Installation type: composer
- Operating System / Distribution & Version: Windows 10

### Current Behavior
PHPMD does not detect errors located within Slim route. PHPMD correctly detects errors on other code within the same file, but not within the Slim route. Here's an example:

```php
<?php

use Slim\Factory\AppFactory;

class TestClass{
    protected $variableNameThatIsTooLong = 4; //rule violation
}

$app = AppFactory::create();
$app->get('blah/blah', function(Request $request, Response $response, $args){
    $variableNameThatIsTooLong = 4; //no rule violation detected
});
```

### Expected Behavior
PHPMD rule violations should be detected within Slim route closures.

### Steps To Reproduce:
Execute PHPMD on the given code, with the LongVariable rule.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by MarkVaughn** [comment]
I'm not able to test this right now but can this be reduced to 

```
<?php
$callback = function(){
    $variableNameThatIsTooLong = 4; //no rule violation detected
};
```

**Comment by james-a-henderson** [comment]
@MarkVaughn I just tried that, and that wasn't working either. I think it has something to do with closures. Unfortunately, even if the solution did work, that would still be an issue. I'm trying to add PHPMD to an existing application, and I'm not going to modify hundreds of existing Slim routes to get linting.

**Comment by kylekatarnls** [comment]
For all the rules, we check only the symbols of the files (functions/classes/traits). Code outside of symbols is not analyzed by the parser.

We sure could improve this to handle procedural code.

But meanwhile, as it's a best practice, please consider putting a maximum of your code into classes. With Slim it's also possible for routes:

```php
<?php

use Slim\Factory\AppFactory;

class BlahController
{
    public function blah(Request $request, Response $response, $args)
    {
        $variableNameThatIsTooLong = 4; //no rule violation detected
    }
}

$app = AppFactory::create();
$app->get('blah/blah', \BlahController::class . ':blah');
```

This will trigger the PHPMD rules and using controller classes will surely improve the code organization and testability.

**Comment by ravage84** [comment]
I think this problem was mentioned before in other issues. Can we (the maintainers) please link this to similar reported issues and then close all issues except the one that reports the problem the best way?

**Comment by kylekatarnls** [comment]
This is probably related to the same root cause than:
#540
#573
